dts/bindings/comparator/nordic,nrf-lpcomp.yaml

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Nordic nRF LPCOMP (analog Low-Power COMParator)
+
+  The following example displays the minimum node layout:
+
+    comp: comp@deadbeef {
+            compatible = "nordic,nrf-lpcomp";
+            reg = <0xdeadbeef 0x1000>;
+            interrupts = <0 NRF_DEFAULT_IRQ_PRIORITY>;
+            status = "disabled";
+    };
+
+  Enabling the comparator node requires setting the default
+  configuration of the comparator.
+
+  The following example displays enabling the comparator
+  using an internal reference:
+
+    &comp {
+            status = "okay";
+            positive-input = "INPUT_0";
+            reference-selection = "SUPPLY_4_8";
+            hysteresis = "ENABLED";
+    };
+
+  To select an external reference, select the "EXT_REF"
+  reference and add the external reference:
+
+    &comp {
+            ...
+            reference-selection = "EXT_REF";
+            external-reference = "REF1";
+            ...
+    };
+
+compatible: "nordic,nrf-lpcomp"
+
+include: base.yaml
+
+properties:
+  positive-input:
+    type: string
+    enum:
+      - "INPUT_0"
+      - "INPUT_1"
+      - "INPUT_2"
+      - "INPUT_3"
+      - "INPUT_4"
+      - "INPUT_5"
+      - "INPUT_6"
+      - "INPUT_7"
+
+  external-reference:
+    type: string
+    enum:
+      - "REF0"
+      - "REF1"
```

Comment:
"REF0" and "REF1" are confusing, "AIN0" and "AIN1" should be used instead. And why not just add these two as values of `reference-selection`, instead having `reference-selection` and `external-reference` separately?